linera-views/src/store.rs

```diff
@@ -255,3 +255,91 @@ pub trait TestKeyValueDatabase: KeyValueDatabase {
         database.open_shared(&[])
     }
 }
+
+/// A module containing a dummy store used for caching views.
+pub mod dummy_store {
+    use super::*;
+
+    /// A dummy store, not actually stored anything - used for caching views.
+    pub struct DummyStore;
+
+    /// A dummy error struct for the dummy store.
+    #[derive(Clone, Copy, Debug)]
+    pub struct DummyStoreError;
+
+    impl std::fmt::Display for DummyStoreError {
+        fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+            write!(f, "dummy error")
+        }
+    }
+
+    impl From<bcs::Error> for DummyStoreError {
+        fn from(_other: bcs::Error) -> Self {
+            Self
+        }
+    }
+
+    impl std::error::Error for DummyStoreError {}
+
+    impl KeyValueStoreError for DummyStoreError {
+        const BACKEND: &'static str = "dummy";
+    }
+
+    impl WithError for DummyStore {
+        type Error = DummyStoreError;
+    }
+
+    impl ReadableKeyValueStore for DummyStore {
```

Comment:
Would it be correct to `panic!` here? to detect calls to something that shouldn't be called? Or maybe even use the `!` (unreachable) type?